custom_components/easee/easee.py

```diff
@@ -46,7 +46,74 @@ async def get_consumption_between_dates(self, from_date, to_date):
         return float(value)
 
     async def start(self):
-        return await self.session.post(f"/api/chargers/{self.id}/commands/start_charging")
+        """Start charging session"""
+        return await self.easee.post(f"/api/chargers/{self.id}/commands/start_charging")
+
+    async def pause(self):
+        """Pause charging session"""
+        return await self.easee.post(f"/api/chargers/{self.id}/commands/pause_charging")
+
+    async def resume(self):
+        """Resume charging session"""
+        return await self.easee.post(
+            f"/api/chargers/{self.id}/commands/resume_charging"
+        )
+
+    async def stop(self):
+        """Stop charging session"""
+        return await self.easee.post(f"/api/chargers/{self.id}/commands/stop_charging")
+
+    async def toggle(self):
+        """Toggle charging session start/stop/pause/resume """
+        return await self.easee.post(
+            f"/api/chargers/{self.id}/commands/toggle_charging"
+        )
+
+    async def get_basic_charge_plan(self):
+        """Get and return charger basic charge plan setting from cloud """
+        return await self.easee.get(
+            f"/api/chargers/{self.id}/commands/basic_charge_plan"
+        )
+
+    async def set_basic_charge_plan(
+        self, id, chargeStartTime, chargeStopTime, repeat=True
+    ):
+        """Set and post charger basic charge plan setting to cloud """
+        json = {
+            "id": id,
+            "chargeStartTime": chargeStartTime,
+            "chargeStopTime": chargeStopTime,
+            "repeat": repeat,
+        }
+        return await self.easee.post(
+            f"/api/chargers/{self.id}/commands/basic_charge_plan", json=json
+        )
+
+    async def delete_basic_charge_plan(self):
+        """Delete charger basic charge plan setting from cloud """
+        return await self.easee.post(
+            f"/api/chargers/{self.id}/commands/basic_charge_plan"
+        )
+
+    async def override_schedule(self):
+        """Override scheduled charging and start charging"""
+        return await self.easee.post(
+            f"/api/chargers/{self.id}/commands/override_schedule"
+        )
+
+    async def smart_charging(self):
+        """Set charger smart charging setting"""
+        return await self.easee.post(f"/api/chargers/{self.id}/commands/smart_charging")
```

Comment:
Have you figured out how this works behind the hood?

Reply:
Not sure since I currently am unable to perform real-life tester on my charger, only desk exercises. User manual refers to a smart charging mode which can also be enabled with the touch button on the charger itself. I think smart_charging() is related to override_schedule(). With the touch button you can enable smart charging but also force it to override smart charging and start immediately. 

Smart charging is described in manual as a method for delay charging to hours of the day where it may be lower price. Not sure if this is limited to some Tibber features or if it is possible with a "plain Eassee". You see the API also refers to partners, and it is possible to partner up with for instance your local grid supplier, so maybe it works differently depending on partner.